Modules/_collectionsmodule.c

```diff
@@ -44,15 +44,24 @@ find_module_state_by_def(PyTypeObject *type)
 /*[clinic input]
 module _collections
 class _tuplegetter "_tuplegetterobject *" "clinic_state()->tuplegetter_type"
+class _collections.deque "dequeobject *" "clinic_state()->deque_type"
 [clinic start generated code]*/
-/*[clinic end generated code: output=da39a3ee5e6b4b0d input=7356042a89862e0e]*/
+/*[clinic end generated code: output=da39a3ee5e6b4b0d input=a033cc2a8476b3f1]*/
+
+typedef struct dequeobject dequeobject;
 
 /* We can safely assume type to be the defining class,
  * since tuplegetter is not a base type */
 #define clinic_state() (get_module_state_by_cls(type))
 #include "clinic/_collectionsmodule.c.h"
 #undef clinic_state
 
+/*[python input]
+class dequeobject_converter(self_converter):
+    type = "dequeobject *"
```

Comment:
Self converters are usually not needed. The only effect of it in this file is that it allows to use a different name for `self`. Alternatively we can just add line
```c
dequeobject *deque = self;
```
at the beginning of every function or simply rename variables.

 Maybe add an option for the `class` directive to specify the name of `self`?

Reply:
> Self converters are usually not needed. The only effect of it in this file is that it allows to use a different name for `self`.

That is not the only effect; the self type is explicitly overridden.

The diff of this PR was made nicer because of this self converter; I think it was a good idea.


(Sorry, I messed up your post, @serhiy-storchaka; I'm at times clumsy with the GitHub UI; it was not my intention.)

Reply:
> Alternatively we can just add line [...] at the beginning of every function [...]

One of the points with Argument Clinic is to _reduce_ boiler plate code ;)

> or simply rename variables.

The diff would have been harder to read; it would have introduced a lot of unneeded churn.

Reply:
Since it requires *adding* boiler plate declaration, it does not make it simpler.

If it could be possible without adding `deque: dequeobject` in every function, it would be the point.